File: day2/python/d02a.py
with open('../../inputs/d02.txt', 'r') as file:
    data = [line.split(" ") for line in file.readlines()]


depth = 0
horizontal = 0

for command in data:
    if command[0] == 'forward':
        horizontal += int(command[1])
    elif command[0] == 'up':
        depth -= int(command[1])
    elif command[0] == 'down':
        depth += int(command[1])

print(depth, horizontal, depth * horizontal)

File: d02a.js
const fs = require('fs')
const readline = require('readline')

const fileLines = []

const fileInterface = readline.createInterface({
    input: fs.createReadStream('./inputs/d02.txt')
})

fileInterface.on('line', line => {
    const [command, value] = line.split(" ")
    fileLines.push({ command, value: +value })
})

fileInterface.on('close', () => {
    const position = {
        horizontal: 0,
        depth: 0
    }

    fileLines.forEach(({ command, value }) => {
        switch (command) {
            case 'forward':
                position.horizontal += value
                break
            case 'up':
                position.depth -= value
                break
            case 'down':
                position.depth += value
        }
    })

    console.log(position, position.horizontal * position.depth)
})

